File: medium/1161. Maximum Level Sum of a Binary Tree/1161. Maximum Level Sum of a Binary Tree.py
from typing import List, Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        queue: List[TreeNode] = [root]
        i = 1
        res = 1, float("-inf")

        while queue:
            level_sum = 0
            level_size = len(queue)

            for _ in range(level_size):
                node = queue.pop(0)

                level_sum += node.val

                if node.left:
                    queue.append(node.left)

                if node.right:
                    queue.append(node.right)

            if level_sum > res[1]:
                res = i, level_sum

            i += 1

        return res[0]

File: medium/1161. Maximum Level Sum of a Binary Tree/1161. Maximum Level Sum of a Binary Tree.go
package main

import "math"

/**
 * Definition for a binary tree node.
 */

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxLevelSum(root *TreeNode) int {
	i := 1
	level := i
	maxSum := math.MinInt

	queue := []*TreeNode{root}

	for len(queue) > 0 {
		levelSum := 0
		levelSize := len(queue)

		for j := 0; j < levelSize; j++ {
			node := queue[0]
			queue = queue[1:]

			levelSum += node.Val

			if node.Left != nil {
				queue = append(queue, node.Left)
			}

			if node.Right != nil {
				queue = append(queue, node.Right)
			}
		}

		if levelSum > maxSum {
			maxSum = levelSum
			level = i
		}

		i += 1
	}

	return level
}
